show-portal-user-info/src/main/java/com/esri/arcgismaps/sample/showportaluserinfo/screens/MainScreen.kt

```diff
@@ -0,0 +1,232 @@
+        }
+    )
+}
+
+/**
+ * Allows the user to enter a url and load a portal.
+ * It uses OAuth under the hood, and has a button to clear credentials.
+ *
+ * @param url the string url to display in the text field
+ * @param onSetUrl called when the url should be changed
+ * @param onSignOut called when any stored credentials should be cleared
+ * @param onLoadPortal called when the [url] should be loaded
+ * @since 200.2.0
+ */
+@OptIn(ExperimentalComposeUiApi::class)
+@Composable
+private fun PortalDetails(
+    url: String,
+    onSetUrl: (String) -> Unit,
+    onSignOut: () -> Unit,
+    onLoadPortal: () -> Unit
+) {
+    Column(
+        modifier = Modifier
+            .fillMaxWidth()
+            .padding(8.dp),
+        horizontalAlignment = Alignment.CenterHorizontally,
+        verticalArrangement = Arrangement.Center
+    ) {
+        // The Url text field
+        OutlinedTextField(
+            modifier = Modifier.fillMaxWidth(),
+            value = url,
+            onValueChange = onSetUrl,
+            label = { Text("Url") },
+            keyboardOptions = KeyboardOptions(
+                keyboardType = KeyboardType.Uri,
+                imeAction = ImeAction.Go
+            ),
+            keyboardActions = KeyboardActions(onAny = { onLoadPortal() }),
+            singleLine = true
+        )
+        Row(
+            modifier = Modifier.fillMaxWidth(),
+            horizontalArrangement = Arrangement.SpaceBetween,
+            verticalAlignment = Alignment.CenterVertically
+        ) {
+            val keyboardController = LocalSoftwareKeyboardController.current
+            val focusManager = LocalFocusManager.current
+            // Clear credential button
+            Button(
+                onClick = {
+                    onSignOut()
+                    keyboardController?.hide()
+                    focusManager.clearFocus()
+                },
+                colors = ButtonDefaults.buttonColors(containerColor = MaterialTheme.colorScheme.tertiary)
+            ) {
+                Text(text = "Sign out")
+            }
+            // Load button
+            Button(
+                onClick = {
+                    onLoadPortal()
+                    keyboardController?.hide()
+                    focusManager.clearFocus()
+                },
+                colors = ButtonDefaults.buttonColors(containerColor = MaterialTheme.colorScheme.primary)
+            ) {
+                Text(text = "Load")
+            }
+        }
+    }
+}
+
+/**
+ * Displays messages to the user. This may be used to display instructions, portal info, or error messages.
+ *
+ * @param infoText the text to display
+ * @param username to display username
+ * @param email to display email
+ * @param creationDate to display creationDate
+ * @param portalName to display portalName
+ * @param userThumbnail to display userThumbnail
+ * @param isLoading whether a progress indicator should be displayed
+ * @since 200.2.0
```

Comment:
The `@param` use in comments is discouraged by the Kotlin [coding conventions](https://kotlinlang.org/docs/coding-conventions.html#documentation-comments):

>Generally, avoid using @param and @return tags. Instead, incorporate the description of parameters and return values directly into the documentation comment, and add links to parameters wherever they are mentioned. Use @param and @return only when a lengthy description is required which doesn't fit into the flow of the main text.

Reply:
Other samples are using @param too, good to have this followed this across all samples in future